### PR TITLE
Eg cohort data documentation

### PR DESCRIPTION
**Pull request by egolol**
cohort data and subclasses

### CODE REVIEW

File: braph2/cohort/Data.m [hunk 1]
```diff
@@ -1,34 +1,115 @@
 classdef Data < handle & matlab.mixin.Copyable
+    % Data (Abstract) A Data
+    % Data provides the methods necesary for all datas
```
Comment (annacanal) [inline braph2/cohort/Data.m:3]: typo: **necessary**

Comment (annacanal) [reply braph2/cohort/Data.m:3]: Also, **data** is plural. Datas does not exist
Suggestion:  necessary for all _data types_

File: braph2/cohort/Data.m [hunk 2]
```diff
@@ -1,34 +1,115 @@
...
         end
     end
     methods  % Basic methods
         function str = tostring(d)
+            % TOSTRING string with information about data
+            %
+            % STRING = TOSTRING(D) returns string with the data class, 
+            % size and value.
+            %
+            % See also disp.
```
Comment (annacanal) [inline braph2/cohort/Data.m:74]: Add a white line after disp.

File: braph2/cohort/Data.m [hunk 3]
```diff
@@ -1,34 +1,115 @@
...
+            % DISP(D) displays information about the data class, size and
+            % value.
+            %
+            % See also tostring.
+            
             disp(['<a href="matlab:help ' Data.getClass(d) '">' Data.getClass(d) '</a>'])
             disp([' size: ' int2str(size(d.getValue(), 1)) ' rows x ' int2str(size(d.getValue(), 2)) ' columns'])
         end
         function atlas = getBrainAtlas(d)
+            % GETBRAINATLAS returns the brain atlas.
```
Comment (annacanal) [inline braph2/cohort/Data.m:89]: delete final point

File: braph2/cohort/Data.m [hunk 4]
```diff
@@ -1,34 +1,115 @@
...
+            %
+            % ATLAS = GETBRAINATLAS(D) returns the brain atlas.
+            %
+            % See also getValue.
+            
             atlas = d.atlas;
         end
         function value = getValue(d)
+            % GETVALUE returns the value of data.
```
Comment (annacanal) [inline braph2/cohort/Data.m:98]: delete final point

File: braph2/cohort/Data.m [hunk 5]
```diff
@@ -1,34 +1,115 @@
...
+            % 
+            % VALUE = GETVALUE(D) returns the value of the data.
+            %
+            % See also getBrainAtlas.
+            
             value = d.value;
         end
         function setBrainAtlas(d, atlas)
+            % SETBRAINATLAS sets the brain atlas to the data object.
```
Comment (annacanal) [inline braph2/cohort/Data.m:107]: delete final point

File: braph2/cohort/Data.m [hunk 6]
```diff
@@ -38,17 +119,30 @@ function setBrainAtlas(d, atlas)
         end
     end
     methods (Abstract)  % setValue
+        % SETVALUE(D, VALUE) (Abstract) sets the value of the data.
+        % Must be used by the sub classes.
```
Comment (annacanal) [inline braph2/cohort/Data.m:123]: subclasses together

File: braph2/cohort/Data.m [hunk 7]
```diff
@@ -38,17 +119,30 @@ function setBrainAtlas(d, atlas)
         end
     end
     methods (Abstract)  % setValue
+        % SETVALUE(D, VALUE) (Abstract) sets the value of the data.
```
Comment (annacanal) [inline braph2/cohort/Data.m:122]: delete final point

File: braph2/cohort/Data.m [hunk 8]
```diff
@@ -38,17 +119,30 @@ function setBrainAtlas(d, atlas)
...
             data_list = subclasses( ...
                 'Data', ...
                 [fileparts(which('Data')) filesep 'datas'] ...
                 );
         end
         function data_class = getClass(d)
-            % data class (same as the data object name)
+            % GETCLASS returns the class of the data 
+            %
+            % DATA_CLASS = GETCLASS(D) reutrns the class of the data.
```
Comment (annacanal) [inline braph2/cohort/Data.m:143]: typo: **returns**

File: braph2/cohort/datas/DataConnectivity.m [hunk 9]
```diff
@@ -1,16 +1,51 @@
 classdef DataConnectivity < Data
+    % DataConnectivity A Data Connectivity Matrix
```
Comment (annacanal) [inline braph2/cohort/datas/DataConnectivity.m:2]: A data connectivity matrix.
Uppercase just at the beginning.

File: braph2/cohort/datas/DataConnectivity.m [hunk 10]
```diff
@@ -1,16 +1,51 @@
...
+    %   setValue                - checks the value and sets it to DataConnectivity
+    %
+    % DataConnectivity static mehtods
+    %   getClass                - returns the class
+    %   getName                 - returns the name of the class
+    %   getDescription          - returns the description of the class
+    %   getAvailableSettings    - returns the available settings
+    %
+    % See also Data, DataFunctional, DataScalar, DataStructural.
     methods
```
Comment (annacanal) [inline braph2/cohort/datas/DataConnectivity.m:24]: space between lines

File: braph2/cohort/datas/DataConnectivity.m [hunk 11]
```diff
@@ -1,16 +1,51 @@
...
-        function d = DataConnectivity(atlas, value)
+        function d = DataConnectivity(atlas, value, varargin)
+            % DATACONNECTIVITY(ATLAS, VALUE, 'PROPERTYVALUE1', 'VALUE1', ...)
+            % creates DataConnectivity object, checks if value is present,
+            % and calls for the super class.
```
Comment (annacanal) [inline braph2/cohort/datas/DataConnectivity.m:28]: superclass together (like subclass)

File: braph2/cohort/datas/DataConnectivity.m [hunk 12]
```diff
@@ -26,15 +61,44 @@ function setValue(d, value)
     end
     methods (Static)
         function data_class = getClass()
+            % GETCLASS returns the class of the data 
+            %
+            % DATA_CLASS = GETCLASS() reutrns the class of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataConnectivity.m:66]: typo: **returns**

File: braph2/cohort/datas/DataConnectivity.m [hunk 13]
```diff
@@ -26,15 +61,44 @@ function setValue(d, value)
...
+            % GETNAME returns the name of the data
+            %
+            % NAME = GETNAME(D) returns the name of the data.
+            % 
+            % See also getClass, getDescription, getAvailableSettings.
+            
             name = 'Connectivity Matrix';
         end
         function description = getDescription()
+            % GETDESCRIPTION returns the description of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataConnectivity.m:82]: delete final point

File: braph2/cohort/datas/DataFunctional.m [hunk 14]
```diff
@@ -1,16 +1,52 @@
 classdef DataFunctional < Data
+    % DataFunctional A Data functional Matrix
```
Comment (annacanal) [inline braph2/cohort/datas/DataFunctional.m:2]: **data** and **matrix**  in lowercase

File: braph2/cohort/datas/DataFunctional.m [hunk 15]
```diff
@@ -1,16 +1,52 @@
...
+    % DataFunctional implements Data and serves as a container for matrix
+    % type data.
+    % It is a subclass of Data.
+    %
+    % DataFunctional implements Data and serves as a container for matrix
+    % type data. It checks if the value of the data being saved is numeric
+    % and has the same number of rows as the Brain Atlas.
+    %
+    % DataFunctional methods:
+    %   DataFunctional        - Constructor
```
Comment (annacanal) [inline braph2/cohort/datas/DataFunctional.m:12]: align - constructor

File: braph2/cohort/datas/DataFunctional.m [hunk 16]
```diff
@@ -1,16 +1,52 @@
...
             end
             
-            d = d@Data(atlas, value);
+            d = d@Data(atlas, value, varargin{:});
         end
     end
     methods
         function setValue(d, value)
+            % SETVALUE sets the value of the data into the object.
+            %
```
Comment (annacanal) [inline braph2/cohort/datas/DataFunctional.m:43]: delete final point

File: braph2/cohort/datas/DataFunctional.m [hunk 17]
```diff
@@ -26,16 +62,45 @@ function setValue(d, value)
     end
     methods (Static)
         function data_class = getClass()
+            % GETCLASS returns the class of the data 
+            %
+            % DATA_CLASS = GETCLASS() reutrns the class of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataFunctional.m:67]: **returns**

File: braph2/cohort/datas/DataFunctional.m [hunk 18]
```diff
@@ -26,16 +62,45 @@ function setValue(d, value)
...
+            %
+            % NAME = GETNAME(D) returns the name of the data.
+            % 
+            % See also getClass, getDescription, getAvailableSettings.
+            
             name = 'Functional Brain Data';
         end
         function description = getDescription()
+            % GETDESCRIPTION returns the description of the data.
+            %
```
Comment (annacanal) [inline braph2/cohort/datas/DataFunctional.m:84]: delete final point

File: braph2/cohort/datas/DataScalar.m [hunk 19]
```diff
@@ -1,17 +1,52 @@
...
+    % DataScalar implements Data and serves as a container for scalar
+    % type data.
+    % It is a subclass of Data.
+    %
+    % DataScalar implements Data and serves as a container for scalar
+    % type data. It checks if the value of the data being saved is numeric
+    % and has the same dimensions as the Brain Atlas.
+    %
+    % DataScalar methods:
+    %   DataScalar        - Constructor
```
Comment (annacanal) [inline braph2/cohort/datas/DataScalar.m:12]: align -constructor

File: braph2/cohort/datas/DataScalar.m [hunk 20]
```diff
@@ -1,17 +1,52 @@
...
             
-            d = d@Data(atlas, value);
+            d = d@Data(atlas, value, varargin{:});
         end
     end
     methods
         function setValue(d, value)
-                        
+            % SETVALUE sets the value of the data into the object.
+            %
```
Comment (annacanal) [inline braph2/cohort/datas/DataScalar.m:42]: delete final point

File: braph2/cohort/datas/DataScalar.m [hunk 21]
```diff
@@ -21,15 +56,44 @@ function setValue(d, value)
...
+            % GETNAME returns the name of the data
+            %
+            % NAME = GETNAME(D) returns the name of the data.
+            % 
+            % See also getClass, getDescription, getAvailableSettings.
+            
             name = 'Scalar Value';
         end
         function description = getDescription()
+            % GETDESCRIPTION returns the description of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataScalar.m:77]: delete final point

File: braph2/cohort/datas/DataStructural.m [hunk 22]
```diff
@@ -1,17 +1,53 @@
 classdef DataStructural < Data
+    % DataStructurral A Data column vector
```
Comment (annacanal) [inline braph2/cohort/datas/DataStructural.m:2]: data lowercase

File: braph2/cohort/datas/DataStructural.m [hunk 23]
```diff
@@ -1,17 +1,53 @@
...
+    % type data.
+    % It is a subclass of Data.
+    %
+    % DataStructurral implements Data and serves as a container for column
+    % type data. It checks if the value of the data being saved is numeric
+    % and has the same number of elements as the Brain Atlas.
+    %
+    % DataStructurral methods:
+    %   DataStructurral        - Constructor
+    %
```
Comment (annacanal) [inline braph2/cohort/datas/DataStructural.m:13]: align

File: braph2/cohort/datas/DataStructural.m [hunk 24]
```diff
@@ -1,17 +1,53 @@
...
             end
             
-            d = d@Data(atlas, value);
+            d = d@Data(atlas, value, varargin{:});
         end
     end
     methods
         function setValue(d, value)
-                        
+            % SETVALUE sets the value of the data into the object.
```
Comment (annacanal) [inline braph2/cohort/datas/DataStructural.m:42]: delete final point

File: braph2/cohort/datas/DataStructural.m [hunk 25]
```diff
@@ -26,16 +62,45 @@ function setValue(d, value)
     end
     methods (Static)
         function data_class = getClass()
+            % GETCLASS returns the class of the data 
+            %
+            % DATA_CLASS = GETCLASS() reutrns the class of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataStructural.m:67]: **returns**

File: braph2/cohort/datas/DataStructural.m [hunk 26]
```diff
@@ -26,16 +62,45 @@ function setValue(d, value)
...
+            % GETNAME returns the name of the data
+            %
+            % NAME = GETNAME(D) returns the name of the data.
+            % 
+            % See also getClass, getDescription, getAvailableSettings.
+            
             name = 'Structural Brain Data';
         end
         function description = getDescription()
+            % GETDESCRIPTION returns the description of the data.
```
Comment (annacanal) [inline braph2/cohort/datas/DataStructural.m:83]: delete final point

File: braph2/cohort/Data.m [hunk 27]
```diff
@@ -1,34 +1,116 @@
 classdef Data < handle & matlab.mixin.Copyable
+    % Data (Abstract) A Data
+    % Data provides the methods necesary for all data types
```
Comment (annacanal) [inline braph2/cohort/Data.m:3]: typo: necessary

File: braph2/cohort/Data.m [hunk 28]
```diff
@@ -1,34 +1,116 @@
 classdef Data < handle & matlab.mixin.Copyable
+    % Data (Abstract) A Data
```
Comment (giovannivolpe) [inline braph2/cohort/Data.m:2]: A data item

File: braph2/cohort/Data.m [hunk 29]
```diff
@@ -1,34 +1,116 @@
...
+    % Data basic methods:
+    %   tostring                - returns a string representing the Data
+    %   disp                    - displays the Data
+    %   value                   - returns the value
+    %   setBrainAtlas           - sets the Brain Atlas to Data
+    % 
+    % Data abstract methods:
+    %   setValue                - method to be inherit by subclass to set the value
+    %
+    % Data static mehtods
```
Comment (giovannivolpe) [inline braph2/cohort/Data.m:21]: add :

**Pull request merged**